code/datums/chatmessage.dm

```diff
@@ -561,6 +658,14 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 	var/duration = BALLOON_TEXT_TOTAL_LIFETIME(duration_mult)
 	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
 
+/datum/chatmessage/proc/approx_str_width(var/string, var/font_size = DEFAULT_FONT_SIZE)
+	var/value = 0
+	for(var/i in 1 to length(string))
+		var/size = letters[string[i]]
+		if(!size)
+			size = 1000
+		value += CHAR_WIDTH(size, font_size)
+	return value
```

Comment:
Is having a for loop on every character actually an optimisation over what its doing now?

Reply:
This is very simple linear loop with access to the assoc list.
The overall cost is O(n), I doubt it's any worse.

Reply:
Faster than MeasureText() for sure.